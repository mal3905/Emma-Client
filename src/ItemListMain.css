
/* for app title */
ul {
    list-style-type: none;
}

.item-list {
    list-style-type: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
    border-top: 4px solid rgba(83, 176, 194, 0.406);
}

.Item__add-item-button {

    text-decoration: none;
    padding: 5px;
    margin-top: 30px;
    background: rgb(232, 163, 175);
    color: black;
    cursor: pointer;
    text-decoration: none;
    font-size: 12px;

   
} 

.ItemList_button-container{
background: red;
}