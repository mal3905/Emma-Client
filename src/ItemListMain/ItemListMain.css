.item-list {
    list-style-type: none;
    display: flex;
    flex-direction: column;
    justify-content: center;
}

.add-btn {
    color: black;
    cursor: pointer;
    text-decoration: none;
    font-size: 30px;
    border: black solid 1px;
    border-radius: 50%;
    padding: 15px;
    margin: 0px 370px 0px 380px;
} 

.item-list a {
    text-decoration: none;
    margin-top: 20px;
}

.home {
    color: black;
    padding: 10px;
    font-size: 30px;
    margin: 100px 350px 0px 350px;
    border: 2px solid black;
    text-decoration: none;
    background-color: transparent;
    list-style-type: none;
}

a:-webkit-any-link {
    text-decoration: none;
    color: black;
}

@media only screen and (min-width: 480px) {
    
}