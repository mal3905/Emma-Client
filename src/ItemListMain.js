
import React from 'react'
import { Link } from 'react-router-dom'
import Item from './Item'
import MyContext from './MyContext'
import { getItemsForCategory } from './item-helpers'
import Button from './Button'
import './ItemListMain.css'


export default class ItemListMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = MyContext

  render() {

    const { categoryid } = this.props.match.params
    const { items=[] } = this.context
    const itemsForCategory = getItemsForCategory(items, categoryid)
    // console.log(itemsForCategory)
    // console.log(items)
    // console.log(categoryid)
    // console.log(this.context.items) 

    return (
      <section className='ItemList'>
        <section>
            <h2>My List </h2>
        </section>
        <ul className='item-list'>
          {itemsForCategory.map(item =>
            <li key={item.id}>
              <Item 
                itemId={item.id}
                name={item.name}
                onDeleteItem={this.handleDeleteItem}
              />
            </li>
          )}
        </ul>
        <div className='ItemList__button-container'>
          <Button 
            tag={Link}
            to='/add-item'
            type='button'
            className='Item__add-item-button'
          >
            add item
          </Button>
        </div>
      </section>
    )
  }
}