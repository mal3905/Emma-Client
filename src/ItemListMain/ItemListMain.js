
import React from 'react'
import { Link } from 'react-router-dom'
import Item from '../Items/Item'
import MyContext from '../contexts/MyContext'
import { getItemsForCategory } from '../item-helpers'
import Button from '../Buttons/Button'
import './ItemListMain.css'


export default class ItemListMain extends React.Component {
  static defaultProps = {
    match: {
      params: {}
    }
  }
  static contextType = MyContext

  render() {

    const { categoryid } = this.props.match.params
    const { items=[] } = this.context
    const itemsForCategory = getItemsForCategory(items, categoryid)
    // console.log(itemsForCategory)
    // console.log(items)
    // console.log(categoryid)
    // console.log(this.context.items) 

    return (
      <section className='ItemList'>
        <div class='home'>
          <Link to={'/'}>Home</Link>
        </div>
        <section>
        </section>
        <ul className='item-list'>
          {itemsForCategory.map(item =>
            <li key={item.id}>
              <Item 
                itemId={item.id}
                name={item.name}
                onDeleteItem={this.handleDeleteItem}
              />
            </li>
          )}
        </ul>
        <div className='add-btn'>
          <Button 
            tag={Link}
            to='/add-item'
            type='button'
            className='Item__add-item-button'
          >
           +
          </Button>
        </div>
      </section>
    )
  }
}