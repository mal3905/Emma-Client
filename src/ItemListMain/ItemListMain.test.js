import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ItemListMain from './ItemListMain'

describe(`ItemListMain component`, () => {
  it('renders a .ItemListMain by default', () => {
    const wrapper = shallow(<ItemListMain />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it.skip('renders Item in ul for each item in array', () => {
    const props = {
      match: {
        params: {
          categoryid: 'THIS_CATEGORY_ID'
        }
      }
    }
    const context = {
      items: [
        {
          "id": "cbc787a0-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "fish",
          "categoryid": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
        
        },
        {
          "id": "d26e0034-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "cherries",
          "categoryid": "THIS_CATEGORY_ID",
    
        },
        {
          "id": "d26e01a6-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "canned soup",
          "categoryid": "THIS_CATEGORY_ID",
         
        },
        {
          "id": "d26e0570-ffaf-11e8-8eb2-f2801f1b9fd1",
          "name": "Peas",
          "categoryid": "b0715efe-ffaf-11e8-8eb2-f2801f1b9fd1",
         
        },
      ]
    }
    const ul = shallow(<ItemListMain {...props} />, context)
      .find('ul')
    expect(toJson(ul)).toMatchSnapshot()
  })
})
