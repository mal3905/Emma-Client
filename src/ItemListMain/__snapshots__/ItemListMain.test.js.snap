// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemListMain component renders a .ItemListMain by default 1`] = `
<section
  className="ItemList"
>
  <ul
    className="item-list"
  />
  <div
    className="add-btn"
  >
    <Button
      className="Item__add-item-button"
      tag={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      to="/add-item"
      type="button"
    >
      Add Item
    </Button>
  </div>
</section>
`;
